Automattic/jetpack
# Subscriptions: Adapt message based on Publicize status

From p1675793272904399-slack-C9EJ7KSGH
Related https://github.com/Automattic/jetpack/issues/28725

## Proposed changes:

Adapts the 'Newsletter' panel message based on whether the post will be shared on social.

**Publicize enabled**
<img width="410" alt="image" src="https://user-images.githubusercontent.com/36432/217848400-9e315654-8c57-46e9-babb-f38d9fac5263.png">

**Publicize disabled**

<img width="434" alt="image" src="https://user-images.githubusercontent.com/36432/217862035-42957e08-69b6-4a65-b5b2-f30069d11554.png">

**Publicize disabled post-publish**

<img width="514" alt="image" src="https://user-images.githubusercontent.com/36432/217849287-18a19892-75de-41c9-988f-e6f558c7beb6.png">


### Other information:


## Jetpack product discussion

See p1675793272904399-slack-C9EJ7KSGH

## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

1. Sandbox a Site with a Business plan.
2. Run `jetpack build plugins/jetpack` and `jetpack rsync jetpack <host>` to sync the change to your sandbox.
3. Remove all Publicize connections and verify the message appears as expected ('This post will reach 1 subscriber and wonʼt be shared on social.')
4. Add a Publicize connection and verify the message appears as expected ('This post will reach 1 subscriber and 482 social followers.').
5. Disable the Publicize connection, publish the post, and verify the message appears as expected ('This post was shared to 1 subscriber.')
